File: algorithms/golang/1-TwoSum/exercise1/two_sum.go
package exercise1 

// 思路分析：
// 假设遍历数组的第一个元素是2，而 target = 9, 所以可以得知另一个值是 9-2 = 7
// 然后去判断数组中是否有 7 这个元素，以此类推
func TwoSum(nums []int, target int) []int {
	for i := 0; i < len(nums); i++ {
		another := target - nums[i]
		for j := i + 1; j < len(nums); j++ {
			if nums[j] == another {
				result := []int{i, j}
				return result
			}
		}
	}
	return nil
}

File: algorithms/java/src/W202213/TwoSum.java
package W202213;

import java.util.HashMap;
import java.util.Map;

/**
 * https://leetcode-cn.com/problems/two-sum/
 *
 * @author tsingyun
 * @version V1.0
 * @date 2022/3/23 23:19
 */
@SuppressWarnings("DuplicatedCode")
public class TwoSum {
	public int[] twoSum(int[] nums, int target) {
		if (nums == null || nums.length == 0) {
			return new int[0];
		}
		Map<Integer, Integer> map = new HashMap<>();
		for (int i = 0; i < nums.length; i++) {
			if (map.get(target - nums[i]) != null) {
				return new int[]{map.get(target - nums[i]), i};
			}
			map.put(nums[i], i);
		}
		return new int[0];
	}
}
